src/components/TheConsumptionParameters.vue

```diff
@@ -68,7 +70,15 @@ function setSavings(savingsPercent: number) {
             :min="1"
             :step="1000"
             :validator="validatePositive"
-            :on-input="emit('consumptionChanged', model.consumption)"
```

Comment:
A single call to `emit()` while rendering the template at least could not have worked.

Reply:
see next comment below